libvcx/src/credential_def.rs

```diff
@@ -43,9 +43,43 @@ pub struct CredentialDef {
 pub struct RevocationDetails {
     pub support_revocation: Option<bool>,
     pub tails_file: Option<String>,
+    pub tails_url: Option<String>,
     pub max_creds: Option<u32>,
 }
 
+#[derive(Deserialize, Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RevocationRegistryDefinitionValue {
+    pub issuance_type: String,
+    pub max_cred_num: u32,
+    pub public_keys: serde_json::Value,
+    pub tails_hash: String,
+    pub tails_location: String
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RevocationRegistryDefinition {
```

Comment:
Nice, it's cool great to have these structures mapped into datastructures 👍 

We could use this to type field `rev_reg_def ` inside existing structure `RevocationRegistry`. We'd just have to remove this `camelCase` annotations (also on `RevocationRegistryDefinitionValue `) and I think it should work out of the box? 

- In that case, `get_tails_hash` in `src/credential_def.rs` could be also simplified and we could rather fail fast during deserialization of credential definition.
- Also function `_replace_tails_location` could then easily be eliminated

Reply:
I don't think this is very practical, as anoncreds requires strings as inputs and also returns strings, which are not modified, except for replacing the tails location.

Reply:
Hmm I see, probably number of de/serializations would even increase. I am not sure how many changes that would be and if it's worth it, but I have a feeling it would be cleaner - I could look at `RevocationRegistry` and directly see it links to `RevocationRegistryDefinition`.
But I don't insist, we can keep as is.